src/reporters/json.ts

```diff
@@ -292,7 +379,7 @@ export default class JSONReporter extends BaseReporter {
 
     this.runner.on(
       'journey:end',
-      ({
+      async ({
```

Comment:
move to sync way, otherwise we would have hard time figuring out when to exactly close the underlying stream. 

Once this is sorted out, I will add tests.